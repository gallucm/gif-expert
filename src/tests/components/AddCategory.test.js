const { shallow } = require("enzyme");
const { AddCategory } = require("../../components/AddCategory");

describe('Pruebas en <AddCategory/>', () => {

    const setCategory = jest.fn();
    let wrapper = shallow(<AddCategory setCategory={setCategory} /> );
    const value = 'Hola Mundo';
    
    beforeEach( () =>{
        jest.clearAllMocks(); // Limpiar mocks o simulaciones anteriores.
        wrapper = shallow(<AddCategory setCategory={setCategory} /> );
    })

    test('Debe mostrarse el componente correctamente', () => {
       expect(wrapper).toMatchSnapshot();
    });

    test('Debe cambiar los valores del input text', () => {
        const input = wrapper.find('input');        
        input.simulate('change', { target: {value} });

        const p = wrapper.find('p').text().trim();

        expect(p).toBe(value);
    });

    test('No debe ejecutar la informaciÃ³n con onSubmit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});        

        expect(setCategory).not.toHaveBeenCalled();
    });

    test('Debe llamar "SetCategory" y limpiar el inputText', () => {
        const wrapper = shallow(<AddCategory setCategory={setCategory} /> );

        wrapper.find('input').simulate('change', {target: {value} });

        let p = wrapper.find('p').text().trim();

        expect(p).toBe(value);

        wrapper.find('form').simulate('submit', {preventDefault(){} });

        expect(setCategory).toHaveBeenCalled();

        const empty = '';

        wrapper.find('input').simulate('change', {target: {empty} });

        p = wrapper.find('p').text().trim();

        expect(p).toBe(empty);        
    });
    
    
})
