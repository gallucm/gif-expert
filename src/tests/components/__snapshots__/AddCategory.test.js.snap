// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> Debe mostrarse el componente correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p>
       
       
    </p>
    <input
      onChange={[Function]}
      placeholder="Busca algo..."
      type="text"
      value=""
    />
    Â 
    <button
      class="btn btn-success btn-search"
      onSubmit={[Function]}
      title="Buscar..."
      type="submit"
    >
      Buscar
    </button>
  </form>
  <br />
</Fragment>
`;
